File: FunctionalProgrammingInFSharp/CSharpCode/Parser.cs


using System;

namespace CSharp
{
    public class Parser
    {
        public static bool TryParseInt(string value, out int i)
        {
            i = 0;

            try
            {
                i = int.Parse(value);
                return true;
            }
            catch (Exception)
            {
                return false;
            }
        }

        public static void ReadInt()
        {
            var line = Console.ReadLine();
            if (TryParseInt(line, out var i))
            {
                Console.WriteLine($"Successfully read integer: {i}");
            }
            else
            {
                Console.WriteLine("Failed to read integer");
            }
        }
    }
}

File: FunctionalProgrammingInFSharp/FunctionalProgrammingInFSharp/Parser.fs


namespace FSharp

open System

module Parser = 
    let TryParseInt value = 
        try
            Some(Int32.Parse value)
        with
        | _ -> None

    let ReadInt() = 
        let line = Console.ReadLine()
        match TryParseInt line with
        | Some i -> printfn "Successfully read integer: %i" i
        | None -> printfn "Failed to read integer"